bd-logger/src/logger.rs

```diff
@@ -468,9 +472,16 @@ impl LoggerHandle {
   pub fn set_feature_flags(&self, flags: Vec<(String, Option<String>)>) {
     with_reentrancy_guard!(
       {
-        let result = AsyncLogBuffer::<LoggerReplay>::set_feature_flags(&self.tx, flags);
-        if let Err(e) = result {
-          log::warn!("failed to set feature flags: {e:?}");
+        for (flag, variant) in flags {
+          let value = variant.unwrap_or_default();
+          let result = self.tx.try_send(AsyncLogBufferMessage::UpsertState(
+            bd_state::Scope::FeatureFlag,
+            flag,
+            value,
+          ));
+          if let Err(e) = result {
+            log::warn!("failed to set feature flag: {e:?}");
+          }
```

Comment:
nit: not your code but these warnings are useless for the most part. We should use unexpected error or just make them warn only in debug builds or something so they get compiled out. I suppose they have some value in customer logs during local development but that's about it.